l10n_it_bill_of_entry/models/account_move.py

```diff
@@ -204,20 +205,34 @@ def _prepare_bill_of_entry_storno(self):
 
     def _reconcile_bill_of_entry_storno(self, move):
         self.ensure_one()
-        reconcile_ids = []
-        for move_line in move.line_ids:
-            line_account = move_line.account_id
-            for boe in self.forwarder_bill_of_entry_ids:
-                boe_payable_lines = boe.line_ids.filtered(
-                    lambda line: line.account_type == "liability_payable"
-                )
-                boe_account = first(boe_payable_lines).account_id
-                if line_account == boe_account:
-                    reconcile_ids.append(move_line.id)
-                    for boe_move_line in boe.line_ids:
-                        if boe_move_line.account_id == boe_account:
-                            reconcile_ids.append(boe_move_line.id)
-        return self.env["account.move.line"].browse(reconcile_ids).reconcile()
+        move_line_ids = move.line_ids.filtered(
+            lambda line: line.account_type == "liability_payable"
+        )
+        if not move_line_ids:
+            raise UserError(_("No storno lines to reconcile"))
+        for move_line in move_line_ids:
+            reconcile_ids = []
+            reconcile_ids.append(move_line.id)
+            boe_line_ids = move_line.bill_of_entry_id.line_ids.filtered(
+                lambda line: line.account_type == "liability_payable"
+            )
+            if not boe_line_ids:
+                raise UserError(_("No bill of entry lines to reconcile"))
+            for boe_line in boe_line_ids:
+                if move_line.account_id == boe_line.account_id:
+                    reconcile_ids.append(boe_line.id)
+                else:
+                    raise UserError(
+                        _(
+                            "Can't reconcile line %(ml_name)s %(ml_account)s "
+                            "with line %(bl_name)s %(bl_account)s",
+                            ml_name=move_line.name,
+                            ml_account=move_line.account_id.name,
+                            bl_name=boe_line.name,
+                            bl_account=boe_line.account_id.name,
+                        )
+                    )
```

Comment:
Filtrare prima le righe per tipo di conto, e poi per conto mi sembra ripetitivo perché il conto giusto avrà per forza il tipo corretto; alla fine a noi interessano solo le righe del conto giusto no?
Cosa ne pensi di questa proposta?
```suggestion
            boe_line_ids = move_line.bill_of_entry_id.line_ids.filtered(
                lambda boe_line, move_line_account=move_line.account_id: boe_line.account_id == move_line_account
            )
            if not boe_line_ids:
                raise UserError(_("No bill of entry lines to reconcile"))
            reconcile_ids.extend(boe_line_ids.ids)
```


Quando viene sollevata questa eccezione poi l'utente cosa dovrebbe fare per correggere? Si potrebbe aggiungere qualche indicazione nell'eccezione?

Reply:
Ci sta come logica e passa i test

Reply:
Visto che l'errore non dovrebbe mai succedere, allora lasciamo perdere le indicazioni da dare all'utente.
 :+1: